src/App/App.xaml.cs

```diff
@@ -173,7 +173,7 @@ private async Task CheckPasswordlessLoginRequestsAsync()
                 DeviceType = loginRequestData.RequestDeviceType
             });
             await _stateService.SetPasswordlessLoginNotificationAsync(null);
-            await Application.Current.MainPage.Navigation.PushModalAsync(new NavigationPage(page));
+            Device.BeginInvokeOnMainThread(async () => await Application.Current.MainPage.Navigation.PushModalAsync(new NavigationPage(page)));
```

Comment:
Use `await Device.InvokeOnMainThreadAsync(...)` instead. It allows bubbling up the exceptions and prevents crashing the app due to the async void code block inside that could potentially throw an exception. Here is the implementation of [InvokeOnMainThreadAsync](https://github.com/xamarin/Xamarin.Forms/blob/5.0.0/Xamarin.Forms.Core/Device.cs#L109) if you want to look at it.